File: 1.bin-test/md5.lua

if (#arg ~= 1) then 
    print(string.format("%s <num>", arg[0]))
    os.exit(1)
end

local md5 = require "md5"
local socket = require "socket"

local fin
local st = socket.gettime()
for var=0, arg[1]-1 do
    fin = md5.sumhexa(var)
end
local et = socket.gettime()

local tc = math.floor((et - st)*1000000)
print(string.format("%s: %d, tv:%d millisecs", fin, arg[1]-1, tc))

File: 1.bin-test/md5.js
var c = require('crypto');
function md5(data) {

    let hash = c.createHash('md5');
    return hash.update(data).digest('hex');
}



function run_md5_n_times_with_timer(num) {
    let st = new Date().getTime();
    let fin = '';
    let n = 0;
    let ns = '';
    for(; n<num; n+=1) {
        ns = `${n}`;
        fin = md5(ns);
    }
    let et = new Date().getTime();
    console.log(`${fin}: ${ns}, tv:${(et-st)} millisecs`);
}

if(process.argv.length !== 3) {
    console.log(`<num>`);
    process.exit(1)
}
run_md5_n_times_with_timer(Number(process.argv[2]));